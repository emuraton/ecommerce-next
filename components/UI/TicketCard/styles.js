import styled, { css } from 'styled-components';
import { ifProp } from 'styled-tools';

const Wrapper = styled.article`
  position: relative;
  background-color: white;
  border-bottom: 1px solid rgb(226, 227, 230);
  padding: 25px 20px;
  border-top: 1px solid rgb(226, 227, 230);

  ${ifProp(
    'notEmpty',
    css`
      &:before,
      &:after {
        position: absolute;
        content: '';
        width: 100%;
        height: 2px;
        top: -2px;
        left: 0;
        background-color: rgb(196, 1, 114);
        z-index: 2;
      }
    `,
  )};
`;

const ColumnContainer = styled.div`
  display: flex;
  justify-content: center;
`;

const H2 = styled.h2`
  color: rgb(36, 37, 41);
  font-size: 16px;
  line-height: 18px;
  margin: 0 0 0 5px;
`;

const Price = styled.p`
  font-weight: 500;
  color: rgb(36, 37, 41);
  margin-bottom: 0px;
  margin-top: 0px;
  font-size: 15px;
  line-height: 18px;
  margin-left: 29px;

  ${ifProp(
    'disabled',
    css`
      font-size: 14px;
      font-weight: 400;
      opacity: 0.5;
      text-decoration: line-through;
    `,
  )};
`;

const TicketDetails = styled.div`
  display: flex;
  align-items: center;
  width: 100%;
  /* border-right: 1px solid rgb(226, 227, 230); */
`;

const Test = styled.div`
  width: 100%;
  /* border-right: 1px solid rgb(226, 227, 230); */
`;

export const Card = {
  Wrapper,
  ColumnContainer,
  H2,
  Price,
  TicketDetails,
  Test,
};
