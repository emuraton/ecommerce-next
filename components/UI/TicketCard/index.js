// @flow
import * as React from 'react';

import PlusMinusBasket from '../PlusMinusBasket';
import Ticket from '../SVG/Ticket';
import { Card } from './styles';

type Item = {
  name: string,
  price: string,
};

type Props = {
  item: Item,
  onClick: (event: SyntheticMouseEvent<any>) => {},
  className?: string,
};

const TicketCard = ({ item, onClick, className }: Props) => {
  if (!item) return null;
  const { name, price } = item;
  return (
    <Card.Wrapper className={className}>
      <Card.ColumnContainer>
        <Card.Test>
          <Card.TicketDetails>
            <Ticket />
            <Card.H2>{name}</Card.H2>
          </Card.TicketDetails>
          <Card.Price>{price}</Card.Price>
        </Card.Test>
        <PlusMinusBasket quantity={0} onClick={onClick} />
      </Card.ColumnContainer>
    </Card.Wrapper>
  );
};

TicketCard.defaultProps = {
  className: null,
};

export default TicketCard;
