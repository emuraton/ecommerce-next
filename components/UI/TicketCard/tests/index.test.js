import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import TicketCard from '..';

describe('<TicketCard />', () => {
  const initialProps = {
    onClick: Function.prototype,
    item: {
      id: 44,
      image: 'shared-room.jpg',
      name: 'Shared room',
      highlights: ['Bunk bed in dormitory', 'Private bathroom', 'Flat screen TV'],
      price: '$18.89',
      previousPrice: '$22',
    },
  };
  describe('@renders', () => {
    it('default render', () => {
      const wrapper = shallow(<TicketCard {...initialProps} />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
