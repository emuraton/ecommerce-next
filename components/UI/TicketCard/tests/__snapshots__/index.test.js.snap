// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TicketCard /> @renders default render 1`] = `
<styles__Wrapper
  className={null}
>
  <styles__ColumnContainer>
    <styles__Test>
      <styles__TicketDetails>
        <Ticket
          className=""
        />
        <styles__H2>
          Shared room
        </styles__H2>
      </styles__TicketDetails>
      <styles__Price>
        $18.89
      </styles__Price>
    </styles__Test>
    <PlusMinusBasket
      onClick={[Function]}
      quantity={0}
    />
  </styles__ColumnContainer>
</styles__Wrapper>
`;
